pkg/controller/sync/controller.go

```diff
@@ -566,54 +533,98 @@ func (s *FederationSyncController) versionName(resourceName string) string {
 	return common.PropagatedVersionName(targetKind, resourceName)
 }
 
-type operationsFunc func(federatedtypes.FederatedTypeAdapter, []string, []string, pkgruntime.Object, pkgruntime.Object, pkgruntime.Object) ([]util.FederatedOperation, error)
-type executionFunc func([]util.FederatedOperation) (map[string]string, []error)
+// syncToClusters ensures that the state of the given object is synchronized to
+// member clusters.
+func (s *FederationSyncController) syncToClusters(clusterNames []string,
+	template, placement, override pkgruntime.Object,
+	propagatedVersion *fedv1a1.PropagatedVersion) reconciliationStatus {
 
-// syncToClusters ensures that the state of the given object is synchronized to member clusters.
-func syncToClusters(operationsAccessor operationsFunc, clusterNames []string, execute executionFunc, adapter federatedtypes.FederatedTypeAdapter, informer util.FederatedInformer, template, placement, override pkgruntime.Object, version *fedv1a1.PropagatedVersion, pendingVersionUpdates sets.String) reconciliationStatus {
-	kind := adapter.Template().Kind()
+	kind := s.adapter.Template().Kind()
 	key := federatedtypes.NewQualifiedName(template).String()
 
 	glog.V(3).Infof("Syncing %s %q in underlying clusters", kind, key)
 
-	selectedClusters, unselectedClusters := computePlacement(adapter.Placement(), placement, clusterNames)
+	selectedClusters, unselectedClusters := computePlacement(s.adapter.Placement(), placement,
+		clusterNames)
+
+	propagatedClusterVersions := getClusterVersions(s.adapter, template, override, propagatedVersion)
 
-	operations, err := operationsAccessor(adapter, selectedClusters, unselectedClusters, template, placement, override)
+	operations, err := s.clusterOperations(selectedClusters, unselectedClusters, template,
+		override, key, propagatedClusterVersions)
 	if err != nil {
+		s.eventRecorder.Eventf(template, corev1.EventTypeWarning, "FedClusterOperationsError",
+			"Error obtaining sync operations for %s: %s error: %s", kind, key, err.Error())
 		return statusError
 	}
 
 	if len(operations) == 0 {
 		return statusAllOK
 	}
 
-	clusterVersions, operationErrors := execute(operations)
+	updatedClusterVersions, operationErrors := s.updater.Update(operations)
 
-	err = updatePropagatedVersion(adapter, clusterVersions, template, override, version, selectedClusters, pendingVersionUpdates)
+	err = updatePropagatedVersion(s.adapter, updatedClusterVersions, template, override,
+		propagatedVersion, selectedClusters, s.pendingVersionUpdates)
 	if err != nil {
-		runtime.HandleError(fmt.Errorf("Failed to record propagated version for %s %q: %v", kind, key, err))
+		runtime.HandleError(fmt.Errorf("Failed to record propagated version for %s %q: %v", kind,
```

Comment:
(No action required) I think linebreaking at a consistent width can hurt readability, as in this case where the change is for the sake of 10 chars.

Reply:
Fair enough. I'm just trying to avoid these hugely long lines that I see all over the k8s code base. I realize there is no standard for the width in k8s, but that really hurts readability IMO.